joblib/test/data/create_numpy_pickle.py

```diff
@@ -30,19 +30,36 @@ def get_joblib_version(joblib_version=joblib.__version__):
     return '.'.join([m.group(1) for m in matches if m is not None])
 
 
-def write_test_pickle(to_pickle):
+def write_test_pickle(to_pickle, args):
+    kwargs = {}
+    compress = args.compress
     joblib_version = get_joblib_version()
     py_version = '{0[0]}{0[1]}'.format(sys.version_info)
     numpy_version = ''.join(np.__version__.split('.')[:2])
-    print('file:', np.__file__)
-    pickle_filename = 'joblib_{0}_compressed_pickle_py{1}_np{2}.gz'.format(
-        joblib_version, py_version, numpy_version)
-    joblib.dump(to_pickle, pickle_filename, compress=True)
-    pickle_filename = 'joblib_{0}_pickle_py{1}_np{2}.pkl'.format(
-        joblib_version, py_version, numpy_version)
-    joblib.dump(to_pickle, pickle_filename, compress=False)
+
+    # The game here is to generate the right filename according to the options.
+    extension = '.gz' if compress else '.pkl'
```

Comment:
Done, but the code is not super nice. At least, one can re-generate the compressed files with it.